mirage/client/dns_client_mirage.ml

```diff
@@ -310,4 +403,13 @@ module Make (R : Mirage_random.S) (T : Mirage_time.S) (M : Mirage_clock.MCLOCK)
   end
 
   include Dns_client.Make(Transport)
+
+  let connect_device ?(protocol= "udp") ?(authenticators= []) ?(nameservers= []) ?(timeout= 5_000_000L) stack =
+    let protocol = match String.lowercase_ascii protocol with
+      | "udp" -> `Udp
+      | "tcp" -> `Tcp
+      | _ -> Fmt.failwith "Invalid protocol: %S" protocol in
+    let authenticators = authenticators_of_strings authenticators in
+    let nameservers = List.map (nameserver_of_string authenticators) nameservers in
+    Lwt.return (create ~nameservers:(protocol, nameservers) ~timeout stack)
```

Comment:
taking a look at the implementation, the `protocol` should defalut to `tcp` (since udp is not supported in this implementation).

Reply:
> (since udp is not supported in this implementation)

I was not aware about that, I believed that both are supported 😕 .